### PR TITLE
Monster behavior tree

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: Infrastructure "Integrate behavior tree implementation into monster AI."

#### Purpose of change
The existing tactical monster AI is kind of a mess of if statements with careful early termination and special casing scattered all over the place, we can do better.

#### Describe the solution
Start integrating the behavior tree implementation into the monster tactical AI.

#### Describe alternatives you've considered
Magic.
I cut it of at essentially a minimum viable implementation for coment and to shakedown the implementation, there's still a significant amount of top-level infrastructure to add.

#### Testing
Added some very abbreviated unit tests.
Need to spawn a shoggoth or other item eating monster and verify that their ability still triggers.

#### Additional context
This is just the first step, hopefully it's fairly clear how this gets extended and it's going to be fairly routine to move the rest of this monster decision making into the behavior tree implementation.

### CODE REVIEW

File: tests/behavior_test.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include "game.h"
```
Comment (kevingranade) [inline tests/behavior_test.cpp:8]:
```suggestion
#include "character_oracle.h"
#include "game.h"
```

**Comment by kevingranade** [comment]
My integration is incomplete, will try to get that fixed tonight.

**Comment by kevingranade** [comment]
Back in business, everything is wired up, shoggoth successfully munched items, but do check behind me.

File: src/monstergenerator.cpp [hunk 2]
```diff
@@ -328,6 +328,14 @@ void load_monster_adjustment( const JsonObject &jsobj )
     adjustments.push_back( adj );
 }
 
+static void build_behavior_tree( mtype &type )
+{
+    type.set_strategy();
+    if( type.has_flag( MF_ABSORBS ) || type.has_flag( MF_ABSORBS_SPLITS ) ) {
```
Comment (ZhilkinSerg) [inline src/monstergenerator.cpp:334]: I guess we will replace flags with goals in monster definition later (or maybe add monster flags to goals definitions)?

**Comment by ZhilkinSerg** [comment]
Shoggoths are still eating items, so that's correct.

File: src/monstergenerator.cpp (hunk 2 above)
Comment (kevingranade) [reply src/monstergenerator.cpp:334]: This should probably get unified around special attacks, or maybe rename them to "abilities", but I haven't thought about that too much yet.
The behavior elements could even be inlined along with metadata about the attack itself.

**Pull request merged**